File: 88-merge-sorted-array/88-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        
        
        int nums1Last = m-1;
        
        int nums2Last = n-1;
        
        int insertAt = m+n-1;
        
        
        while(nums2Last >= 0){
            
            if(nums1Last >= 0 && nums1[nums1Last] > nums2[nums2Last]){
                
                nums1[insertAt] = nums1[nums1Last];
                insertAt--;
                nums1Last--;
                
            }else{
                
                nums1[insertAt] = nums2[nums2Last];
                nums2Last--;
                insertAt--;
                
            }
            System.out.println(Arrays.toString(nums1) + " " + nums1Last + " " + nums2Last + " "+ insertAt);
        }
        
        
    }
}

File: 88-merge-sorted-array/88-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        
        
        num1Last = m-1
        
        num2Last = n-1
        
        insertPosi = m+n-1
        
        while num2Last >= 0:
            
            if num1Last >= 0 and nums1[num1Last] > nums2[num2Last]:
                nums1[insertPosi] = nums1[num1Last]
                insertPosi -= 1
                num1Last -= 1
            else:
                nums1[insertPosi] = nums2[num2Last]
                num2Last -= 1
                insertPosi -= 1
        
        